File: app/src/main/java/com/user/canopas/dagger/di/component/ApiComponent.java
package com.user.canopas.dagger.di.component;

import com.user.canopas.dagger.MainActivity;
;
import com.user.canopas.dagger.di.Module.ApiModule;
import com.user.canopas.dagger.di.Scope.CustomScope;

import dagger.Component;

@CustomScope
@Component(modules = ApiModule.class,dependencies = NetworkComponent.class)
public interface ApiComponent {

    void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/user/canopas/dagger/di/component/ApiComponent.kt
package com.user.canopas.dagger.di.component

import com.user.canopas.dagger.MainActivity
import com.user.canopas.dagger.di.Module.ApiModule
import com.user.canopas.dagger.di.Scope.CustomScope

import dagger.Component

@CustomScope
@Component(modules = arrayOf(ApiModule::class), dependencies = arrayOf(NetworkComponent::class))
interface ApiComponent {

    fun inject(mainActivity: MainActivity)
}
